presto-jdbc/src/main/java/io/prestosql/jdbc/PrestoParameterMetaData.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.jdbc;
+
+import com.google.common.collect.ImmutableList;
+
+import java.sql.ParameterMetaData;
+import java.sql.SQLException;
+import java.sql.Types;
+import java.util.List;
+
+import static io.prestosql.jdbc.ColumnInfo.getType;
+import static java.util.Objects.requireNonNull;
+
+public class PrestoParameterMetaData
+        implements ParameterMetaData
+{
+    private final List<ParameterInfo> parameterInfo;
+
+    PrestoParameterMetaData(List<ParameterInfo> parameterInfo)
+    {
+        this.parameterInfo = ImmutableList.copyOf(requireNonNull(parameterInfo, "parameterInfo is null"));
+    }
+
+    @Override
+    public int getParameterCount() throws SQLException
+    {
+        return parameterInfo.size();
+    }
+
+    @Override
+    public int isNullable(int param) throws SQLException
+    {
+        throw new NotImplementedException("ParameterMetaData", "isNullable");
```

Comment:
We need to implement all of these methods, otherwise programs that use the driver will fail. For nullable, it's fine to always return `columnNullableUnknown`. Using `ColumnInfo` as @Praveen2112 suggested should make this easier, as we'll have some of this information.